### PR TITLE
Update trace campaigns after editing givethio

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
And Refactor adminBro codes
related to Giveth/giveth-next#173, Giveth/giveth-next#363, Giveth/giveth-next#362

This PR and https://github.com/Giveth/feathers-giveth/pull/618 should merge together

### CODE REVIEW

File: entities/project.ts [hunk 1]
```diff
@@ -76,6 +76,14 @@ class Project extends BaseEntity {
   @Column({ nullable: true })
   description?: string;
 
+  @Field({ nullable: true })
+  @Column({ nullable: true })
+  traceCampaignId?: string;
+
+  @Field(type => Boolean)
+  @Column({ default: false })
+  isTraceable: boolean;
```
Comment (aminlatifi) [inline entities/project.ts]: This field is redundant, the existence of `traceCampaignId` implies traceability of the project

File: resolvers/projectResolver.ts [hunk 2]
```diff
@@ -422,6 +423,8 @@ export class ProjectResolver {
     project.qualityScore = qualityScore;
     await project.save();
 
+    // We dont wait for trace reponse, because it may increase our response time
+    updateCampaignInTrace(project);
```
Comment (aminlatifi) [inline resolvers/projectResolver.ts]: Change the name like trace please, a name like `dispatchCampaignUpdateEvent`

File: services/trace/traceService.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+export interface UpdateCampaignData {
+  title?: string;
+  campaignId?: string;
+  description?: string;
+  verified?: boolean;
+  archived?: boolean;
+}
+
+export const updateCampaignInTrace = async (
```
Comment (aminlatifi) [inline services/trace/traceService.ts]: Please update the name like trace, a name like `emitCampaignUpdateEvent`

File: services/trace/traceService.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+          projectId: project.id,
+        },
+      );
+      return;
+    }
+    const payload: UpdateCampaignData = {
+      campaignId: project.traceCampaignId,
+      title: project.title,
+      description: project.description,
+      verified: project.verified,
```
Comment (aminlatifi) [inline services/trace/traceService.ts:49]: Can verification change here?

**Review by aminlatifi — CHANGES_REQUESTED** [review]
Thanks @mohammadranjbarz

File: services/trace/traceService.ts (hunk 4 above)
Comment (mohammadranjbarz) [reply services/trace/traceService.ts:49]: Yeah, admin may change it in admin bro panel

**Comment by aminlatifi** [comment]
LGTM!
Thanks @mohammadranjbarz

**Pull request merged**